roles/kubernetes-apps/metrics_server/templates/metrics-server-deployment.yaml.j2

```diff
@@ -91,12 +92,14 @@ spec:
           effect: NoSchedule
 {% endif %}
       affinity:
-        nodeAffinity:
+        podAntiAffinity:
           preferredDuringSchedulingIgnoredDuringExecution:
           - weight: 100
```

Comment:
Thanks @ugur99 
Is the "weight: 100" and "podAffinityTerm" required , ref to the "https://github.com/kubernetes-sigs/metrics-server/blob/master/manifests/components/high-availability/patch.yaml"

Reply:
Thank you for the review @yankay, 

Yes, I checked the suggested HA deployment yml, but I didn't want to enforce the user with "requiredDuringSchedulingIgnoredDuringExecution"; instead of it, I preferred to use "preferredDuringSchedulingIgnoredDuringExecution" for podAntiAffinity.

Reply:
Thanks @ugur99 .

The metrics-server is not pure 'stateless' , So I'm not sure about which ( required or preferred ) is better . Is it possible we submit a PR/Issue to the metrics-server repo , and has a discuss with the metrics-server team future.

Thank you very much 👍

Reply:
Hi @yankay, thanks for the comments. I submitted a PR to [metrics_server](https://github.com/kubernetes-sigs/metrics-server/pull/1148) as you suggested.